src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -226,17 +224,47 @@ def acr_update_set(client,
     return client.update(resource_group_name, registry_name, parameters)
 
 
-def acr_login(registry_name, resource_group_name=None, username=None, password=None):
+def _try_aad_token_login(login_server):
+    try:
+        username = "00000000-0000-0000-0000-000000000000"
+        password = get_login_refresh_token(login_server)
+        return username, password
+    except Exception as e:  # pylint: disable=broad-except
```

Comment:
Can you narrow the type of the Exception to avoid broad-except violation?

Reply:
I guess since this is one fallback step, if due to whatever reason it failed, the next fallback step will be tried, so it doesn't matter what the exception is. Is it necessary to narrow the exception type here?

Reply:
There is a reason for the `broad except` pylint rule. It may hide issues which you should handle differently or a catastrophe error which have been exposed. Broadly capture all types of exception without a sound understanding of the reason why will lead to potential ambiguous errors in the future.